### PR TITLE
These are some things i spotted while i was refactoring send logs primary key to be ksuid

### PR DESCRIPTION
**Pull request by gudgl**
# Pull Request Template

## Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: consumers/campaigner/main.go [hunk 1]
```diff
@@ -110,8 +110,9 @@ func (h *MessageHandler) HandleMessage(m *nsq.Message) error {
 					"campaign_id": campaign.ID,
 					"status":      campaign.Status,
 				}).Error("unable to update campaign")
-			return nil
+			return err
```
Comment (Dzalevski) [inline consumers/campaigner/main.go:113]: @gudgl we should not return err here since the campaign status is already set to sending.

Comment (gudgl) [reply consumers/campaigner/main.go:113]: @Dzalevski agreed

**Comment by FilipNikolovski** [comment]
@gudgl can you pull this one first https://github.com/mailbadger/app/pull/1078 and then we can merge them together.

**Comment by gudgl** [comment]
@FilipNikolovski @Dzalevski i pulled #1078  you can check it

**Review by FilipNikolovski — COMMENTED** [review]
The refactor looks good to me, can we do a round of tests to make sure everything works correctly?

**Comment by gudgl** [comment]
Basically i added to continue to the next subscriber if one of the following errors are thrown
 - fails to get metadata
 - fails to get unsubscribe url 
 - fails to render template
 - fails to marshal sender params
 - fails to publish message
 I tested it and a received the mail but only with just one email sent to me.

Funny thing when get campaign failed this notification was received
image
and then i received this error
image

Another bad thing for now is if the starting campaign fails we dont know why it failed. We have only log about that.

**Comment by FilipNikolovski** [comment]
> Funny thing when get campaign failed this notification was received
> image
> and then i received this error
> image
> 
> Another bad thing for now is if the starting campaign fails we dont know why it failed. We have only log about that.

That event is not important, we'll add a fix to log that message and not try to unmarshal it.

**Review by FilipNikolovski — APPROVED** [review]

**Pull request merged**